src/udev/udev-event.c

```diff
@@ -549,7 +549,11 @@ static int on_spawn_io(sd_event_source *s, int fd, uint32_t revents, void *userd
 
         l = read(fd, p, size - 1);
         if (l < 0) {
-                if (errno != EAGAIN)
+                if (errno == EAGAIN)
+                        if (sd_event_source_set_enabled(s, SD_EVENT_ONESHOT) < 0)
+                                log_device_error(spawn->device,
+                                                 "Failed to reactivate IO source of '%s'", spawn->cmd);
```

Comment:
please include error code in msg, i.e. change to:

```c
r = sd_event_source_set_enabled(s, SD_EVENT_ONESHOT);
if (r < 0) 
        return log_device_error_errno(spawn->device, r, "Failed to reactivate IO source of '%s': %m", spawn->cmd);
``

Reply:
I have done this in the updated PR